scripts/utils/fuels/relayCommonMessage.ts

```diff
@@ -0,0 +1,140 @@
+      let coins = (await relayer.getCoins()).filter((coin) => coin.assetId == ZeroBytes32 && coin.amount.gt(minGas));
+      if (coins.length == 0) throw new Error('wallet has no single UTXO that can cover gas costs');
+      let gas_coin = coins[0];
+
+      // get contract id
+      const data = arrayify(message.data);
+      if (data.length < 32) throw new Error('cannot find contract ID in message data');
+      const contractId = hexlify(data.slice(0, 32));
+
+      // build the transaction
+      const transaction = new ScriptTransactionRequest({ script, gasLimit: minGas, ...txParams });
+      transaction.inputs.push({
+        type: InputType.Message,
+        amount: message.amount,
+        sender: message.sender.toHexString(),
+        recipient: message.recipient.toHexString(),
+        witnessIndex: 0,
+        data: message.data,
+        nonce: message.nonce,
+        predicate: predicate,
+      });
+      transaction.inputs.push({
+        type: InputType.Contract,
+        txPointer: ZeroBytes32,
+        contractId: contractId,
+      });
+      transaction.inputs.push({
+        type: InputType.Coin,
+        id: gas_coin.id,
+        owner: hexlify(gas_coin.owner.toBytes()),
+        amount: gas_coin.amount,
+        assetId: ZeroBytes32,
+        txPointer: ZeroBytes32,
+        witnessIndex: 0,
+      });
+      transaction.outputs.push({
+        type: OutputType.Contract,
+        inputIndex: 1,
+      });
+      transaction.outputs.push({
+        type: OutputType.Change,
+        to: hexlify(gas_coin.owner.toBytes()),
+        assetId: ZeroBytes32,
+      });
+      transaction.outputs.push({
+        type: OutputType.Variable,
+      });
+      transaction.witnesses.push('0x');
+
+      debug('-------------------------------------------------------------------');
+      debug(transaction.inputs);
+      debug('-------------------------------------------------------------------');
+      debug(transaction.outputs);
+      debug('-------------------------------------------------------------------');
+
+      return transaction;
+    },
+  },
+];
+type CommonMessageDetails = {
+  name: string;
+  predicateRoot: string;
+  predicate: string;
+  script: string;
+  buildTx: (
+    relayer: FuelWallet,
+    message: Message,
+    details: CommonMessageDetails,
+    txParams: Pick<TransactionRequestLike, 'gasLimit' | 'gasPrice' | 'maturity'>
+  ) => Promise<ScriptTransactionRequest>;
+};
+
+// Relay commonly used messages with predicates spendable by anyone
+export async function relayCommonMessage(
+  relayer: FuelWallet,
+  message: Message,
+  txParams: Pick<TransactionRequestLike, 'gasLimit' | 'gasPrice' | 'maturity'> = {}
+): Promise<TransactionResponse> {
+  // find the relay details for the specified message
+  let messageRelayDetails: CommonMessageDetails = null;
+  for (let details of COMMON_RELAYABLE_MESSAGES) {
+    if (details.predicateRoot == message.recipient.toHexString()) {
```

Comment:
Would it be better to factor out the hex call here so that we only do it once instead of every iteration?